pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -180,12 +181,60 @@ func (cfg Config) toAuth3Options() tokens3.AuthOptions {
 	}
 }
 
+// This allows acceptance testing against an existing OpenStack
+// install, using the standard OS_* OpenStack client environment
+// variables.
+func configFromEnv() (cfg Config, ok bool) {
+	cfg.Global.AuthUrl = os.Getenv("OS_AUTH_URL")
+
+	cfg.Global.TenantId = os.Getenv("OS_TENANT_ID")
+	// Rax/nova _insists_ that we don't specify both tenant ID and name
+	if cfg.Global.TenantId == "" {
+		cfg.Global.TenantName = os.Getenv("OS_TENANT_NAME")
```

Comment:
can delete the code, the following will set cfg.Global.TenantId and cfg.Global.TenantId